### PR TITLE
BUGFIX handle files with no extension as if they are invalid

### PR DESCRIPTION
**Pull request by andrewandante**
Fixes https://github.com/silverstripe/silverstripe-framework/issues/9037

Struggling to get the tests to run locally but I've added one in, hopefully Travis can determine that this works.

Aims to gracefully skip over a file if it has no extension rather than implode the task.

### CODE REVIEW

File: src/Dev/Tasks/FileMigrationHelper.php [hunk 1]
```diff
@@ -257,8 +257,14 @@ protected function migrateFile($base, File $file, $legacyFilename)
             return false;
         }
 
-        // Fix file classname if it has a classname that's incompatible with its extention
+        // Ignore extensionless files
         $extension = $file->getExtension();
+        if ($extension === "") {
```
Comment (michalkleiner) [inline src/Dev/Tasks/FileMigrationHelper.php]: Possibly use apostrophes instead of quotes for consistency?

Comment (andrewandante) [reply src/Dev/Tasks/FileMigrationHelper.php]: I only used doubles as the phpdoc for `File::get_file_extension()` shows:

```
 * Examples:
 * - "myfile" returns ""
```

So not too bothered, I guess we aren't interpolating a string so doubles aren't required

File: src/Dev/Tasks/FileMigrationHelper.php [hunk 2]
```diff
@@ -569,4 +561,30 @@ protected function getFilenameArray()
...
     }
+
+    /**
+     * @param File $file
+     * @param array $messages
+     * @param string $legacyFilename
+     *
+     * return void
+     */
+    protected function handleInvalidFile($file, $messages, $legacyFilename)
```
Comment (emteknetnz) [inline src/Dev/Tasks/FileMigrationHelper.php:572]: does this need to be a protected? could we make it private instead and changes this to a patch release? (optional for patch release, would require a rebase/retarget to `1.6`)

**Review by emteknetnz — CHANGES_REQUESTED** [review]
Looks great - one thing re the protected method

Happy to merge after that

File: src/Dev/Tasks/FileMigrationHelper.php (hunk 2 above)
Comment (andrewandante) [reply src/Dev/Tasks/FileMigrationHelper.php:572]: Yeah making it private should be no problem. Happy to point it at 1.6, will have a go this afternoon.

**Pull request merged**